service/frontend/http_api_server.go

```diff
@@ -65,6 +73,9 @@ type HTTPAPIServer struct {
 	serveMux               *runtime.ServeMux
 	stopped                chan struct{}
 	matchAdditionalHeaders map[string]bool
+	nexusHandler           http.Handler
+	metricsHandler         metrics.Handler
+	auth                   *authorization.Policy
```

Comment:
Is this Nexus specific? Maybe `NewHTTPAPIServer` constructor should just accept `additionalPaths map[string]http.Handler` and Nexus can be configured completely outside of this file.

Otherwise, it feels Nexus has invaded/infected this code instead of just making this existing code extensible for other HTTP paths (including Nexus and whatever else in the future).